Wrappers/Python/cil/framework/framework.py

```diff
@@ -3273,6 +3273,83 @@ def get_slice(self,channel=None, vertical=None, horizontal_x=None, horizontal_y=
         else:
             return ImageData(out.array, deep_copy=False, geometry=geometry_new, suppress_warning=True)                            
 
+
+    def apply_circular_mask(self, radius=0.99, in_place=True):
+        """
+
+        Apply a circular mask to the horizontal_x and horizontal_y slices. Values outside this mask will be set to zero.
+
+        This will most commonly used to mask edge artefacts from standard CT reconstructions with FBP.
+
+        Parameters
+        ----------
+        radius : float, default 0.99
+            radius of mask by percentage of size of horizontal_x or horizontal_y, whichever is greater
+
+        in_place : boolean, default True
+            If `True` masks the current data, if `False` returns a new `ImageData` object.
+            
+
+        Returns
+        -------
+        ImageData
+            If `in_place = False` returns a new ImageData object with the masked data
+
+        """
+        ig = self.geometry
+
+        # grid
+        y_range = (ig.voxel_num_y-1)/2
+        x_range = (ig.voxel_num_x-1)/2
+
+        Y, X = numpy.ogrid[-y_range:y_range+1,-x_range:x_range+1]
+        
+        # use centre from geometry in units distance to account for aspect ratio of pixels
+        dist_from_center = numpy.sqrt((X*ig.voxel_size_x+ ig.center_x)**2 + (Y*ig.voxel_size_y+ig.center_y)**2)
+
+        size_x = ig.voxel_num_x * ig.voxel_size_x
+        size_y = ig.voxel_num_y * ig.voxel_size_y
+
+        if size_x > size_y:
+            radius_applied =radius * size_x/2
+        else:
+            radius_applied =radius * size_y/2
+
+        # voxel area = 1, sphere of area=1 has r = 0.56
```

Comment:
Could you elaborate in comments what is happening in the following lines - what is r, why this value, what does scale -1->+1 mean, what does clip(-r,r) do, why divide by r. Any reference/source on the anti-aliasing approximation?

Reply:
`r` is the radius of a circular voxel of the same area as a square voxel. Voxels that have `centres < mask_radius - r` are fully inside the mask and have a value set to `-r`, voxels with their `centres > mask_radius + r` are fully outside the mask and have values set to `r`. Voxel centres that fall in to this border region values now are between `-r` and `r` and give us an approximation of the linear distance of the voxel centre from the defined mask. However we still need to estimate a proportion of the circular voxel to assign inside the circle.

I then divide by `r` this just stretches the values  from -1 to 1, then I then times this by `pi/2` so we have a data range  `-pi/2` - `pi/2` with these border voxels having values between the two.  I take the `sin` of the mask, which is the approximation of the proportion of the border voxel to assign in to the mask region, and then rescale the data from 0-1.

No references, it's a pretty crude approximation but something is better than nothing here!

Reply:
Does that make sense? Do the new comments give a better hint? It's a bit much to write the full thing there!

Reply:
It makes great sense, thanks for the explanation!  I really would add the explanation in comments - it will be very helpful if returning to this code (especially someone else) at a later stage and I'd much rather have a long and clear explanation than a short unclear one :)

Reply:
I think the new comments are good!